test/evergreen.yml

```diff
@@ -432,6 +432,23 @@ functions:
           # Just let the system take a breath
           sleep 10s
         done
+  "upload many-coll artifact":
```

Comment:
It looks like most of the contents in this function are the same as the base `upload artifact` function, except for the `target` and `source_dir` parameters. Can we extend the base function by accepting function parameters so that those duplicated lines could be saved? Another reason I prefer using the same base function is `upload artifact` is currently hooked as a `post` function, which means it would get triggered for all tasks defined in this Evergreen project as a post-tasks command. Calling a customized function won't bypass the calling of the post-tasks function.

Reply:
Sure, I will try this approach.